src/go/rpk/pkg/cli/cluster/storage/status-mount.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package storage
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newMountStatus(fs afero.Fs, p *config.Params) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "status-mount [MIGRATION ID]",
+		Short:   "Status of a mount/unmount operation on a topic to the Redpanda Cluster from Tiered Storage",
+		Long:    `Status of a mount/unmount operation on a topic to the Redpanda Cluster from Tiered Storage.`,
```

Comment:
Can we shorten the message so it fits 80 chars please?

Maybe Short could be:
```
Status of mount/unmount operation
```

As this will be seen from `rpk cluster storage` there is enough context for the command. Then you can expand in the long text 

Also, same with the cancel-mount, we should add a small line to let the user know how to get the Migration ID.